scipy/special/tests/test_faddeeva.py

```diff
@@ -57,3 +57,21 @@ def test_symmetry(self):
             rtol=1e-15,
             atol=0
         )
+    
+    def test_corner_cases(self):
+        points = np.array([
```

Comment:
So the trick with this test data is that it doesn't appear to have a canonical source. That makes sense, since the implementation for these edge cases is simple, but it makes it unclear if the test is correct. I would suggest instead testing for some edge cases with exact outputs (like the `np.inf` entry below). You can do that inside a

```python
@pytest.mark.parametrize('x, sigma, gamma, desired', [
    (0, 0, 0, inf),
    (1, 0, 0, 0),
    ....
])
def test_corner_cases(self, x, sigma, gamma, desired):
    assert sc.voigt_profile(x, sigma, gamma) == desired
```

Then I would add tests that check the _continuity_ as `gamma -> 0` e.g. So compare the result at `gamma = 1e-16` (say) with the result at `gamma = 0` with all other inputs held constant. (And do similar checks for the other limits.)

Reply:
And add tests for `NaN` inputs.